tests/test_cli_compile.py

```diff
@@ -1063,6 +1140,40 @@ def test_annotate_option(pip_conf, runner, option, expected):
     assert out.exit_code == 0
 
 
+@pytest.mark.parametrize(
+    ("input_filename", "input_content"),
+    (
+        ("requirements.in", "small-fake-a==0.1"),
+        (
+            "setup.py",
+            (
+                "from setuptools import setup\n"
+                "setup(name='fake-setuptools-a', install_requires=['small-fake-a==0.1'])"
+            ),
+        ),
+    ),
+)
+def test_annotation_relative_paths(
+    pip_conf, runner, tmp_path, input_filename, input_content
```

Comment:
Per https://github.com/jazzband/pip-tools/pull/1329#issuecomment-802022955 this should fix the issue:
```suggestion
    pip_with_index_conf, runner, tmp_path, input_filename, input_content,
```
But I'm not sure if it's desirable to make this test hit PyPY. @atugushev may have a better idea.

Reply:
Thanks, yeah I'd also rather avoid the network if we can. I notice that some other setuptools-using tests use `@pytest.mark.network`, but `test_relative_local_package` (added in this PR) doesn't, it just uses `pip_conf`:

```ini
[global]
no-index = true
find-links = /home/andy/Code/pip-tools/tests/test_data/minimal_wheels
```

And it seems to do alright...

Reply:
Also, `pip_with_index_conf` doesn't solve the problem anyway:

```ini
[global]
index-url = http://example.com
find-links = /home/andy/Code/pip-tools/tests/test_data/minimal_wheels
```

> which don't have docstrings but their names are pretty verbose.

Haha I guess `verbosity != clarity`

Reply:
> Haha I guess `verbosity != clarity`

Yep. Also, the docstrings are useful since they show up in the `pytest --fixtures` output.

Reply:
Resolving this to reduce noise, as we've got it working. 

I do want to understand it better, though, and will continue that discussion from my latest top-level comment.